autonomous/shoot_move_shoot.py

```diff
@@ -0,0 +1,126 @@
+import math
+
+from wpilib import controller
+from wpilib import geometry
+from wpilib import trajectory
+from magicbot import AutonomousStateMachine, state
+
+from components.chassis import Chassis
+from components.indexer import Indexer
+from components.shooter import Shooter
+from controllers.shooter import ShooterController
+
+
+def to_pose(x: float, y: float, heading: float) -> geometry.Pose2d:
+    """
+    Convert inputs into a wpilib pose object
+    """
+    rotation = geometry.Rotation2d(heading)
+    return geometry.Pose2d(x, y, rotation)
+
+
+class ShootMoveShootBase(AutonomousStateMachine):
+
+    shooter_controller: ShooterController
+
+    chassis: Chassis
+    indexer: Indexer
+    shooter: Shooter
+
+    TARGET_POSITION = geometry.Pose2d(0, -2.404, geometry.Rotation2d(0))
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.controller = controller.RamseteController()
+        self.start_pose = to_pose(0, 0, math.pi)
+        tolerance = to_pose(0.1, 0.1, math.pi / 18)
+        self.controller.setTolerance(tolerance)
+        self.trajectory_config = trajectory.TrajectoryConfig(
+            maxVelocity=1, maxAcceleration=1
+        )
+        self.gen = trajectory.TrajectoryGenerator()
+
+    def setup(self):
+        self.trajectory_config.setKinematics(self.chassis.kinematics)
+        self.trajectory_num = 0
+        self.trajectory_max = 1
```

Comment:
Given we override `__init__` here, we should probably move things that don't depend on the injected variables to it.